impl/src/backend.test.ts

```diff
@@ -0,0 +1,152 @@
+import { fairlyAllocateCredit } from "./backend";
+
+import { strict as assert } from "assert";
+import test from "node:test";
+import * as simplestatistics from "simple-statistics";
+
+interface FairlyAllocateCreditTestCase {
+  name: string;
+  credit: number[];
+  value: number;
+  needsRand?: boolean;
+}
+
+function noRand(): number {
+  throw new Error("no rand expected");
+}
+
+type Interval = [min: number, max: number];
+
+// https://en.wikipedia.org/wiki/Normal_distribution#Quantile_function
```

Comment:
https://en.wikipedia.org/wiki/Probit seems clearer to me.  It defines $\mathbb{\Phi}^{-1}$ further down.